lib/perso.ml

```diff
@@ -2615,141 +2603,73 @@ and eval_person_field_var conf base env ((p, p_auth) as ep) loc = function
       | Vint cnt -> VVstring (string_of_int cnt)
       | _ -> VVstring "")
   | [ "cous_paths_min_date"; l1; l2 ] ->
-      let max_a_l =
-        match get_env "max_anc_level" env with
-        | Vint i -> i
-        | _ -> max_anc_level_default
-      in
-      let max_d_l =
-        match get_env "max_desc_level" env with
-        | Vint i -> i
-        | _ -> max_desc_level_default
-      in
       let _cousins_cnt, cousins_dates =
         match (!Cousins.cousins_t, !Cousins.cousins_dates_t) with
         | Some t, Some d_t -> (t, d_t)
-        | _, _ -> Cousins.init_cousins_cnt base max_a_l max_d_l p
+        | _, _ -> Cousins.init_cousins_cnt conf base p
       in
       let i = try int_of_string l1 with Failure _ -> raise Not_found in
       let j = try int_of_string l2 with Failure _ -> raise Not_found in
-      let min, _ = cousins_dates.(i).(j) in
+      let min, _ =
+        if
+          i + 1 > Array.length cousins_dates
+          || j + 1 > Array.length cousins_dates.(i)
+        then (-1, -1)
+        else cousins_dates.(i).(j)
+      in
       VVstring (string_of_int min)
   | [ "cous_paths_max_date"; l1; l2 ] ->
-      let max_a_l =
-        match get_env "max_anc_level" env with
-        | Vint i -> i
-        | _ -> max_anc_level_default
-      in
-      let max_d_l =
-        match get_env "max_desc_level" env with
-        | Vint i -> i
-        | _ -> max_desc_level_default
-      in
       let _cousins_cnt, cousins_dates =
         match (!Cousins.cousins_t, !Cousins.cousins_dates_t) with
         | Some t, Some d_t -> (t, d_t)
-        | _, _ -> Cousins.init_cousins_cnt base max_a_l max_d_l p
+        | _, _ -> Cousins.init_cousins_cnt conf base p
       in
       let i = try int_of_string l1 with Failure _ -> raise Not_found in
       let j = try int_of_string l2 with Failure _ -> raise Not_found in
-      let _, max = cousins_dates.(i).(j) in
+      let _, max =
+        if
+          i + 1 > Array.length cousins_dates
+          || j + 1 > Array.length cousins_dates.(i)
+        then (-1, -1)
+        else cousins_dates.(i).(j)
+      in
       VVstring (string_of_int max)
   | [ "cous_paths_cnt_raw"; l1; l2 ] -> (
-      let max_a_l =
-        match get_env "max_anc_level" env with
-        | Vint i -> i
-        | _ -> max_anc_level_default
-      in
-      let max_d_l =
-        match get_env "max_desc_level" env with
-        | Vint i -> i
-        | _ -> max_desc_level_default
-      in
-      let l = Cousins.cousins_l1_l2_aux base max_a_l max_d_l l1 l2 p in
+      let l = Cousins.cousins_l1_l2_aux conf base l1 l2 p in
```

Comment:
It would be nice to have `cousins_l1_l2_aux` stay in Cousins too

Reply:
It already is in Cousins !!

Reply:
I mean not visible outside cousins.ml (not in cousins.mli)